deeptime/markov/msm/tram/_tram_model.py

```diff
@@ -0,0 +1,224 @@
+        r""" The estimated logarithm of the lagrange multipliers :math:`v_i^k` of each Markov state for all thermodynamic
+            states. lagrangian_mult_log[k,i] contains the lagrange multiplier of Markov state i in thermodynamic state k.
+        """
+        return self._lagrangian_mult_log
+
+    @property
+    def modified_state_counts_log (self):
+        r"""The logarithm of the modified state counts :math:`R_i^k` of each Markov state for all thermodynamic
+        states. modified_state_counts_log[k,i] contains the state counts of Markov state i in thermodynamic state k.
+        """
+        return self._modified_state_counts_log
+
+    @property
+    def therm_state_energies(self) -> np.ndarray:
+        r""" The estimated free energy per thermodynamic state, :math:`f_k`, where :math:`k` is the thermodynamic state
+        index.
+        """
+        return self._therm_state_energies
+
+
+    @property
+    def markov_state_energies(self) -> np.ndarray:
+        r""" The estimated free energy per Markov state, :math:`f^i`, where :math:`i` is the Markov state
+        index.
+        """
+        return self._markov_state_energies
+
+    @property
+    def markov_state_model_collection(self):
+        r""" The underlying MarkovStateModelCollection. Contains one Markov state model for each sampled thermodynamic
+        state.
+
+        Returns
+        -------
+        markov_state_model_collection : MarkovStateModelCollection
+            the collection of markov state models containing one model for each thermodynamic state.
+        """
+        return self._markov_state_model_collection
+
+    def compute_sample_weights(self, dtrajs, bias_matrices, therm_state=-1):
+        r""" Compute the sample weight :math:`\mu(x)` for all samples :math:`x`. If the thermodynamic state index is >=0,
+        the sample weights for that thermodynamic state will be computed, i.e. :math:`\mu^k(x)`. Otherwise, this gives
+        the unbiased sample weights.
+
+        Parameters
+        ----------
+        therm_state : int, optional
+        The index of the thermodynamic state in which the sample weights need to be computed. If therm_state=-1,
+        the unbiased sample weights are computed.
+
+        Returns
+        -------
+        sample_weights : np.ndarray
+        The statistical weight :math:`\mu(x)` of each sample (i.e., the probability distribution over all samples:
+        the sum over all sample weights equals one.)
+
+        Notes
+        -----
+        The statistical distribution is given by
+
+        .. math:: \mu(x) = \left( \sum_k R^k_{i(x)} \mathrm{exp}[f^k_{i(k)}-b^k(x)] \right)^{-1}
+        """
+        return tram.compute_sample_weights(therm_state, dtrajs, bias_matrices, self._therm_state_energies,
+                                           self._modified_state_counts_log)
+
+    def compute_observable(self, dtrajs, bias_matrices, observable_values, therm_state=-1):
```

Comment:
I would put `observable_values` as first parameter and can we make `dtrajs` and `bias_matrices` optional? So that the original sample weights are used by default

Reply:
yeah alright that is fair. still though it seems more natural to me to put the observable values as first parameter (as in you are evaluating `o(x, *args)`)